data-preprocessing/video_download/nsmq_vid.py

```diff
@@ -0,0 +1,59 @@
+#This python script reads video links from a given sheet in an excel document, 
+#downloads the videos and saves the videos to a google drive folder
+#K.T Yeboah | 27-02-2023| 11:44am
+
+
+#import relevant libaries
+import pandas as pd
+from pytube import YouTube
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaFileUpload
+from pathlib import Path
+
+
+#Define doc variables
+excel_document_name = 'NSMQ.xlsx'  #document name
+excel_sheet_name = 'Sheet1'   #name of sheet in document
+videos_header_name = 'Links' #name of header of column containing video links
+
+# Read the Excel sheet and extract video links
+df = pd.read_excel(excel_document_name, excel_sheet_name)
+video_links = df[videos_header_name]
+
+
+# Download videos and save them to a local directory
+local_directory = 'videos/'
+
+for link in video_links:
+    try:
+        yt = YouTube(link)
+        video = yt.streams.filter(adaptive=True, file_extension='mp4').order_by('resolution').desc().first()
+        video.download(local_directory)
+    except Exception as e:
+        print(f'Error downloading {link}: {e}')
+
+
+# Authenticate with Google Drive API and create a folder to store the videos
+creds = service_account.Credentials.from_service_account_file('credentials.json')
```

Comment:
`credentials.json` should also be another value that is passed to the script and not hardcoded here
also validate what is passed to ensure it contains what you want. if not throw an error

Reply:
> `credentials.json` should also be another value that is passed to the script and not hardcoded here also validate what is passed to ensure it contains what you want. if not throw an error

Sure on this. As a command line argument? or I should allow the data curation team use an instruction file which will contain all necessary input data. Then I read and validate. Just in case the number of cmd line args become many

Reply:
depends on the contents of the `credentials.json` file. if it's many items then is should be a file that is passed in as a command line argument. would be good to use the python argparse approach for your command line arguments: https://docs.python.org/3/library/argparse.html and https://realpython.com/command-line-interfaces-python-argparse/ are  good resources on it